### PR TITLE
Feature - make Stacked Area outlines optional

### PR DESCRIPTION
**Pull request by DalerAsrorov**
## Description

Resolves: https://github.com/eventbrite/britecharts/issues/670

## Motivation and Context

https://github.com/eventbrite/britecharts/issues/670

## How Has This Been Tested?

Added multiple tests

## Screenshots (if appropriate):
screen shot 2018-10-21 at 3 23 34 pm

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/stacked-area.js [hunk 1]
```diff
@@ -833,6 +835,12 @@ define(function(require){
                     .style('stroke', ({key}) => categoryColorMap[key]);
             }
 
+            if (!hasOutline) {
```
Comment (Golodhros) [inline src/charts/stacked-area.js]: Have you think about moving the conditional to the line 819?

File: src/charts/stacked-area.js [hunk 2]
```diff
@@ -1197,6 +1205,15 @@ define(function(require){
             return this;
         };
 
+        exports.hasOutline = function(_x) {
+            if (!arguments.length) {
+                return hasOutline;
+            }
+            hasOutline = _x;
```
Comment (sennab) [inline src/charts/stacked-area.js]: Could you explain what the `_x` parameter is?

File: src/charts/stacked-area.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Golodhros) [reply src/charts/stacked-area.js]: Hahaha, it seemed it worked, just not fast enough!

File: src/charts/stacked-area.js (hunk 2 above)
Comment (Golodhros) [reply src/charts/stacked-area.js]: _x is an optional parameter that if passed, will be set as the new ´hasOutline´value, if not, then this function will return the current ´hasOutline´ variable value.

This pattern is effectively a way of having a getter and a setter all at once.
_x could be called anything different for sure, this is the convention we adopted though.

Comment (sennab) [reply src/charts/stacked-area.js]: Thanks for the explanation! Makes sense, I hadn't seen that convention before.

File: src/charts/stacked-area.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DalerAsrorov) [reply src/charts/stacked-area.js]: I tried moving those to line `819` but the outline still shows up a little bit in animated one.

[the remark this answers is not in the document]
Comment (Golodhros) [reply src/charts/stacked-area.js]: OK, thanks for trying!

**Pull request merged**